### PR TITLE
Update Ganache to v7.x

### PR DESCRIPTION
**Pull request by eggplantzzz**
This PR salvages work done in https://github.com/trufflesuite/truffle/pull/3569 which updates Truffle to use Ganache v7.x. Thanks to @davidmurdoch! 👍

### CODE REVIEW

**Comment by haltman-at** [comment]
So are we closing #3569, then?

**Comment by eggplantzzz** [comment]
Yes I believe we can, @davidmurdoch are you okay with this?

**Comment by THRIVEbluegooglecheck** [comment]
0=0xDb06322ceA4D7e56f538eB54E907ab469d7878E2

**Comment by THRIVEbluegooglecheck** [comment]
=0

File: packages/debugger/test/data/returnvalue.js [hunk 1]
```diff
@@ -279,8 +287,9 @@ describe("Return value decoding", function () {
     try {
       await instance.fail(); //web3 throws on failure
     } catch (error) {
-      txHash = error.hashes[0]; //it's the only hash involved
+      txHash = error.receipt.transactionHash;
     }
+    if (!txHash) assert.fail("should have errored and set txHash");
```
Comment (haltman-at) [inline packages/debugger/test/data/returnvalue.js:292]: Given that we're using Chai, I think it would make more sense to just write this line as
```assert.isDefined(txHash, "should have errored and set txHash")```
(That also applies throughout this file, I won't tag each individual one.)

File: packages/debugger/test/solidity.js [hunk 2]
```diff
@@ -327,8 +335,9 @@ describe("Solidity Debugging", function () {
       try {
         await instance.run(); //this will throw because of the revert
       } catch (error) {
-        txHash = error.hashes[0]; //it's the only hash involved
+        txHash = error.receipt.transactionHash;
       }
+      if (!txHash) assert.fail("should have errored and set txHash");
```
Comment (haltman-at) [inline packages/debugger/test/solidity.js:340]: Another assertion to rewrite.

File: packages/debugger/test/stacktrace.js [hunk 3]
```diff
@@ -126,8 +134,9 @@ describe("Stack tracing", function () {
     try {
       await instance.run(0); //this will throw because of the revert
     } catch (error) {
-      txHash = error.hashes[0]; //it's the only hash involved
+      txHash = error.receipt.transactionHash;
     }
+    if (!txHash) assert.fail("should have thrown and set txHash");
```
Comment (haltman-at) [inline packages/debugger/test/stacktrace.js:139]: And another one of these; there's more in this file of course but I won't mark them each individually.

File: packages/debugger/test/txlog.js [hunk 4]
```diff
@@ -376,8 +384,9 @@ describe("Transaction log (visualizer)", function () {
     try {
       await instance.testRevert(); //this will throw because of the revert
     } catch (error) {
-      txHash = error.hashes[0]; //it's the only hash involved
+      txHash = error.receipt.transactionHash;
     }
+    if (!txHash) assert.fail("should have errored and set txHash");
```
Comment (haltman-at) [inline packages/debugger/test/txlog.js:389]: And another one of these.

File: packages/decoder/test/current/test/downgrade-test.js [hunk 5]
```diff
@@ -579,12 +579,15 @@ async function runErrorTestBody(mangledCompilations) {
...
   //we need the raw return data, and contract.call() does not exist yet,
   //so we're going to have to use web3.eth.call()
-
-  let data = await web3.eth.call({
-    to: deployedContract.address,
-    data: selector
-  });
-
+  let data;
+  try {
```
Comment (haltman-at) [inline packages/decoder/test/current/test/downgrade-test.js:583]: Huh... I'm confused why this change is necessary even though VM errors are off.  What's going on here...? (@davidmurdoch?)

File: packages/decoder/test/current/test/wire-test.js [hunk 6]
```diff
@@ -1072,26 +1070,30 @@ describe("Over-the-wire decoding", function () {
...
+
       //we need the raw return data, and contract.call() does not exist yet,
       //so we're going to have to use web3.eth.call()
-  
-      let data = await web3.eth.call({
-        to: deployedContract.address,
-        data: selector
-      });
+      let data;
+      try {
```
Comment (haltman-at) [inline packages/decoder/test/current/test/wire-test.js:1086]: Same question here as previous comment, naturally.  (And also later in this file but I won't mark them all.)

File: packages/interface-adapter/test/fabric-evm-getId.test.ts [hunk 7]
```diff
@@ -29,18 +34,24 @@ async function prepareGanache(
...
 
-describe("fabric-evm getId Overload", function() {
-  it("returns networkID as valid string instead of number w/ fabric-evm=true", async function() {
-    const preparedGanache = await prepareGanache(true);
-    const networkID = await preparedGanache.interfaceAdapter.getNetworkId();
-    assert(typeof networkID === "string");
-    await preparedGanache.server.close();
+describe("fabric-evm getId Overload", function () {
+  it("returns networkID as valid string instead of number w/ fabric-evm=true", async function () {
+    const preparedGanache = (await prepareGanache(true)) as any;
```
Comment (haltman-at) [inline packages/interface-adapter/test/fabric-evm-getId.test.ts:39]: Why is `as any` necessary here?  (And also below, of course.)  Especially because I notice you didn't use it in the other file...

File: packages/interface-adapter/test/quorum-getBlock.test.ts [hunk 8]
```diff
@@ -33,23 +37,29 @@ async function prepareGanache(
...
   });
 
-  it("recovers block timestamp as number w/ quorum=false", async function() {
-    const preparedGanache = await prepareGanache(false);
-    const block = await preparedGanache.interfaceAdapter.getBlock(0);
-    const expectedBlockTime = new BN(genesisBlockTime.getTime()).divn(1000);
-    assert.strictEqual(block.timestamp, expectedBlockTime.toNumber());
-    await preparedGanache.server.close();
+  it("recovers block timestamp as number w/ quorum=false", async function () {
+    const preparedGanache = (await prepareGanache(false)) as any;
```
Comment (haltman-at) [inline packages/interface-adapter/test/quorum-getBlock.test.ts:56]: Another unnecessary-seeming `as any`.

File: packages/truffle/test/scenarios/migrations/dryrun.js [hunk 9]
```diff
@@ -5,35 +5,27 @@ const assert = require("assert");
...
+  before(async function () {
     this.timeout(10000);
     config = await sandbox.create(project);
     config.network = "development";
     config.logger = logger;
     config.mocha = {
       reporter: new Reporter(logger)
     };
-
-    const provider = new Web3.providers.HttpProvider("http://localhost:8545", {
```
Comment (haltman-at) [inline packages/truffle/test/scenarios/migrations/dryrun.js:28]: So was all this stuff just not used, I'm guessing?

File: packages/truffle/test/scenarios/migrations/errors.js [hunk 10]
```diff
@@ -25,12 +23,6 @@ describe("migration errors", function () {
     config.mocha = {
       reporter: new Reporter(logger)
     };
-
-    const provider = new Web3.providers.HttpProvider("http://localhost:8545", {
```
Comment (haltman-at) [inline packages/truffle/test/scenarios/migrations/errors.js:29]: Same question here, this is all unused?

File: packages/contract/lib/manual-send.js [hunk 11]
```diff
@@ -44,7 +44,11 @@ async function sendTransactionManual(web3, params, promiEvent) {
 
 function handleError(error) {
   debug("error: %O", error);
-  if (error.data && Object.keys(error.data).length === 3) {
+  if (error.data && error.data.hash) {
```
Comment (haltman-at) [inline packages/contract/lib/manual-send.js:47]: Thanks for getting this!

File: packages/contract/lib/reason.js [hunk 12]
```diff
@@ -13,7 +13,13 @@ const reason = {
   _extract: function(res, web3, _interfaceAdapter) {
     //I'm not sure why interfaceAdapter is here if it's not used,
     //so I just put an underscore in front of its name for now...
-    if (!res || (!res.error && !res.result)) return;
+    if (!res) return;
+    if (!res.error && (
```
Comment (haltman-at) [inline packages/contract/lib/reason.js:17]: This logic doesn't look correct to me; and if it somehow is correct, then I'm pretty sure it's been written in a way that's overcomplicated.

This conditional is trying to detect if there's *no* error, right?  So it ought to be checking that there is *both* no 2.x error *and* no 7.x error.  But instead it uses an `||`.  Moreover, it first checks `!res.error`, and then checks `res.error`... that can't be right.  (And if it *is* right, it probably ought to be able to use some Boolean algebra to simplify that.)

File: packages/contract/lib/reason.js [hunk 13]
```diff
@@ -23,14 +29,28 @@ const reason = {
       res && typeof res === "object" && typeof res.result === "string";
 
     if (isObject) {
+      // TODO: ganache 2.0 returns the reason string already when
```
Comment (haltman-at) [inline packages/contract/lib/reason.js:32]: You've left a `TODO` comment in your code, @gnidan isn't going to like that. :P  So, uh, you should probably figure out whether you want to include that in this PR, or just convert this to a non-TODO comment about how this would be a good thing to do in the future.

File: packages/contract/lib/reason.js [hunk 14]
```diff
@@ -23,14 +29,28 @@ const reason = {
       res && typeof res === "object" && typeof res.result === "string";
 
     if (isObject) {
```
Comment (haltman-at) [inline packages/contract/lib/reason.js:31]: Does this condition -- which is perhaps named a little misleadingly -- still apply in the Ganache 7 case?

File: packages/contract/lib/reason.js [hunk 15]
```diff
@@ -23,14 +29,28 @@ const reason = {
...
+      // vmErrorsOnRPCResponse === true. Why not use it?
       const data = res.error.data;
-      const hash = Object.keys(data)[0];
+      let resData;
+      if ("result" in data) {
+        // there is a single result (only in ganache 3.0+)
+        resData = data.result;
+      } else {
+        // handle `evm_mine`, `miner_start`, batch payloads, and ganache 2.0
+        // TODO: there are 1 or more failed transactions, handle multiple?
```
Comment (haltman-at) [inline packages/contract/lib/reason.js:41]: Another `TODO` comment. :P  Either handle multiple or just make a note to handle them later!

**Review by haltman-at — COMMENTED** [review]
OK, most of this seems good, but I have a bunch of questions on a few parts.

File: packages/debugger/test/data/global.js [hunk 16]
```diff
@@ -160,7 +160,13 @@ describe("Globally-available variables", function () {
     provider = Ganache.provider({
       seed: "debugger",
       gasLimit: 7000000,
-      _chainId: 1337 //temporary until Ganache v3!
+      hardfork: "istanbul",
```
[the remark this answers is not in the document]
Comment (haltman-at) [reply packages/debugger/test/data/global.js:163]: (Btw, my preferred approach for this would be to actually update this test to remove the hack, but uh I think it's fine to leave that in for now and I can remove it later. :P )

File: packages/contract-tests/test/deploy.js [hunk 17]
```diff
@@ -1,11 +1,17 @@
...
-  var Example;
-  var web3;
-  var providerOptions = { vmErrorsOnRPCResponse: false };
-
+  let Example;
+  let web3;
+  // ganache v2 had a default blockGasLimit of 0x6691b7, which these tests rely on
+  const providerOptions = {
+    gasLimit: 0x6691b7,
+    hardfork: "istanbul",
```
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/contract-tests/test/deploy.js:10]: @davidmurdoch do you know why Istanbul? I think this might have been something you added?

File: packages/contract-tests/test/errors.js [hunk 18]
```diff
@@ -1,9 +1,13 @@
...
-var util = require("./util");
+const assert = require("chai").assert;
+const util = require("./util");
 
 describe("Client appends errors (vmErrorsOnRPCResponse)", function () {
-  var Example;
-  var providerOptions = { vmErrorsOnRPCResponse: true }; // <--- TRUE
+  let Example;
+  const providerOptions = {
+    hardfork: "istanbul",
```
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/contract-tests/test/errors.js:7]: @davidmurdoch

File: packages/contract-tests/test/errors.js [hunk 19]
```diff
@@ -83,34 +87,26 @@ describe("Client appends errors (vmErrorsOnRPCResponse)", function () {
...
         assert(
-          e.stack.includes("Error: base fee exceeds gas limit"),
+          e.stack.includes("Error: intrinsic gas too low"),
           "Should keep hijacked error description"
         );
         assert(
           e.stack.includes("/test/errors.js:"),
           "Should include original stack details"
         );
-        assert(
```
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/contract-tests/test/errors.js:99]: To be honest I don't really know why these were here in the first place but after discussion it seemed like something we didn't care about (maybe we did at one point). I probably should remove the other ones as well, these were just the ones that caused tests to fail.

File: packages/core/lib/commands/develop/run.js [hunk 20]
```diff
@@ -51,9 +52,9 @@ module.exports = async options => {
     mnemonic,
     gasLimit: customConfig.gas || 0x6691b7,
     gasPrice: customConfig.gasPrice || 0x77359400,
-    noVMErrorsOnRPCResponse: true,
     time: config.genesis_time,
-    _chainId: 1337 //temporary until Ganache v3!
+    vmErrorsOnRPCResponse: customConfig.vmErrorsOnRPCResponse || false,
```
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/core/lib/commands/develop/run.js:56]: Ah you are probably right that this should be left off until later.

File: packages/debugger/test/context-vyper.js [hunk 21]
```diff
@@ -42,19 +42,28 @@ const sources = {
...
+      miner: {
+        instamine: "strict"
+      }
+    });
   });
 
   before("Prepare contracts and artifacts", async function () {
     this.timeout(30000);
-
+    console.log("in prepare ----");
```
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/debugger/test/context-vyper.js:64]: Indeed :)

File: packages/contract-tests/test/errors.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/contract-tests/test/errors.js:99]: I removed a couple of assertions involving directly checking `e.hijackedStack`. I left in the tests whose descriptions reference hijacked stack because some of them appear like they could still be relevant.

File: packages/contract-tests/test/errors.js [hunk 22]
```diff
@@ -236,14 +235,16 @@ describe("Client appends errors (vmErrorsOnRPCResponse)", function () {
       const example = await Example.new(1);
       try {
         const accounts = await Example.web3.eth.getAccounts();
-        await example.runsOutOfGas({ from: accounts[0], gas: "0x6691b7" });
+        // specifying a gasLimit as ganache @2.7.0 no longer limits gas
```
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/contract-tests/test/errors.js:238]: It seems like there used to be no gas limit and now in order to ensure it runs out of gas in this way we specifcy it manually?

File: packages/debugger/test/data/global.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/debugger/test/data/global.js:163]: David suggested that it is likely here because these tests might rely on `gasPrice = 0`.

File: packages/debugger/test/data/returnvalue.js (hunk 1 above)
Comment (eggplantzzz) [reply packages/debugger/test/data/returnvalue.js:292]: Sure!

File: packages/contract-tests/test/errors.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (davidmurdoch) [reply packages/contract-tests/test/errors.js:7]: istanbul allows running with a `0` gasPrice

File: packages/contract-tests/test/deploy.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (davidmurdoch) [reply packages/contract-tests/test/deploy.js:10]: istanbul allows running with a `0` gasPrice

File: packages/truffle/test/scenarios/migrations/dryrun.js (hunk 9 above)
Comment (eggplantzzz) [reply packages/truffle/test/scenarios/migrations/dryrun.js:28]: yep there was a bunch of extra stuff

File: packages/truffle/test/scenarios/migrations/errors.js (hunk 10 above)
Comment (eggplantzzz) [reply packages/truffle/test/scenarios/migrations/errors.js:29]: yup

File: packages/debugger/test/endstate.js [hunk 23]
```diff
@@ -33,42 +33,51 @@ uint x;
...
     //does not presently work)
     let txHash;
     try {
-      await instance.run(); //this will throw because of the revert
+      await instance.run();
     } catch (error) {
-      txHash = error.hashes[0]; //it's the only hash involved
+      txHash = error.receipt.transactionHash;
     }
+    if (!txHash) assert.fail("it should have set the txHash");
```
Comment (lsqproduction) [inline packages/debugger/test/endstate.js:78]: here's another reformat to `assert.isDefined(txHash, "should have errored and set txHash");` ?

File: packages/deployer/test/ens.js [hunk 24]
```diff
@@ -20,7 +20,7 @@ let ganacheOptions,
 describe("ENS class", () => {
   before(() => {
     ganacheOptions = {
-      // note that when vmErrorsOnRPCResponse is true, eager instamine must be enabled (default)
+      // note that when vmErrorsOnRPCResponse is true, strict instamine must be enabled (default)
```
Comment (davidmurdoch) [inline packages/deployer/test/ens.js:23]: `vmErrorsOnRPCResponse: true` only works with `instamine: "eager"` (which is the default)

File: packages/contract-tests/test/deploy.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (haltman-at) [reply packages/contract-tests/test/deploy.js:10]: Huh, no idea why that's something we want here, but I'll take your word for it.  Is it worth adding a comment here saying that in the future this should be updated not to use zero `gasPrice` so this can be moved off Istanbul in the future?

File: packages/contract-tests/test/errors.js (hunk 22 above)
[the remark this answers is not in the document]
Comment (haltman-at) [reply packages/contract-tests/test/errors.js:238]: OK but then you mean Ganache **7** no longer limits gas, right?

File: packages/debugger/test/data/global.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (haltman-at) [reply packages/debugger/test/data/global.js:163]: Oh, these tests don't rely on gas price being zero; they do, however, currently rely on base fee being zero (until the hack is removed and the tests are properly updated :) ).  So, like, if you're going to pin a specific hardfork, you could at least use Berlin rather than Instabul. :)

File: packages/contract-tests/test/errors.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (gnidan) [reply packages/contract-tests/test/errors.js:7]: (Should we also put a comment here?)

File: packages/contract-tests/test/errors.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (gnidan) [reply packages/contract-tests/test/errors.js:99]: (The other ones have also been removed)

[the remark this answers is not in the document]
Comment (haltman-at) [reply packages/contract-tests/test/errors.js:99]: OK, seems these have been removed now.

File: packages/contract-tests/test/errors.js [hunk 25]
```diff
@@ -236,14 +235,16 @@ describe("Client appends errors (vmErrorsOnRPCResponse)", function () {
...
+        // estimations to the default block gas limit.
```
Comment (gnidan) [inline packages/contract-tests/test/errors.js:239]: Maybe either remove this comment or reword it to be independent of history (we can say _why_ we specify `gasLimit`... seems better than commenting on the past)

File: packages/debugger/test/data/global.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (gnidan) [reply packages/debugger/test/data/global.js:163]: Let's make sure we don't forget the follow-on here... maybe open an issue?

File: packages/contract-tests/test/deploy.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (gnidan) [reply packages/contract-tests/test/deploy.js:10]: Putting this catch-all issue here https://github.com/trufflesuite/truffle/issues/4633

File: packages/contract-tests/test/errors.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (gnidan) [reply packages/contract-tests/test/errors.js:7]: Putting this catch-all issue here https://github.com/trufflesuite/truffle/issues/4633

File: packages/debugger/test/data/global.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (gnidan) [reply packages/debugger/test/data/global.js:163]: Issue made here https://github.com/trufflesuite/truffle/issues/4633

File: packages/decoder/test/current/test/downgrade-test.js (hunk 5 above)
Comment (gnidan) [reply packages/decoder/test/current/test/downgrade-test.js:617]: See papertrail: https://github.com/trufflesuite/ganache/issues/1496

Comment (gnidan) [reply packages/decoder/test/current/test/downgrade-test.js:617]: We probably want this papertrail as a comment, actually! Leaving this unresolved until we get a comment :)

Comment (gnidan) [reply packages/decoder/test/current/test/downgrade-test.js:617]: @cds-amal please resolve when you feel this is sufficiently documented

File: packages/decoder/test/current/test/wire-test.js (hunk 6 above)
Comment (gnidan) [reply packages/decoder/test/current/test/wire-test.js:1086]: @cds-amal same here :)

File: packages/contract/lib/reason.js (hunk 12 above)
Comment (gnidan) [reply packages/contract/lib/reason.js:17]: Let's undo this change and instead borrow @haltman-at's change in https://github.com/trufflesuite/truffle/pull/4506.

File: packages/contract/lib/reason.js [hunk 26]
```diff
@@ -13,7 +13,13 @@ const reason = {
...
     //I'm not sure why interfaceAdapter is here if it's not used,
     //so I just put an underscore in front of its name for now...
-    if (!res || (!res.error && !res.result)) return;
+    if (!res) return;
+    if (!res.error && (
+        !res.result /* ganache 2.0 */
+        || (res.error && res.error.data && !res.error.data.result /* ganache 7.x */)
+      )
+    ) return;
+
```
Comment (gnidan) [inline packages/contract/lib/reason.js:22]:
```suggestion
    if (!res || (!res.error && !res.result)) return;
```

File: packages/contract/lib/reason.js [hunk 27]
```diff
@@ -23,14 +23,28 @@ const reason = {
       res && typeof res === "object" && typeof res.result === "string";
 
     if (isObject) {
+      // TODO: ganache 2.0 returns the reason string already when
+      // vmErrorsOnRPCResponse === true. Why not use it?
```
Comment (gnidan) [inline packages/contract/lib/reason.js:27]:
```suggestion
      // NOTE that Ganache >=2 returns the reason string when
      // vmErrorsOnRPCResponse === true, which this code could
      // be updated to respect (instead of computing here)
```

File: packages/contract/lib/reason.js [hunk 28]
```diff
@@ -23,14 +23,28 @@ const reason = {
...
       const data = res.error.data;
-      const hash = Object.keys(data)[0];
+      let resData;
+      if ("result" in data) {
+        // there is a single result (only in ganache 3.0+)
+        resData = data.result;
+      } else {
+        // handle `evm_mine`, `miner_start`, batch payloads, and ganache 2.0
+        // TODO: there are 1 or more failed transactions, handle multiple?
```
Comment (gnidan) [inline packages/contract/lib/reason.js:35]:
```suggestion
        // handle `evm_mine`, `miner_start`, batch payloads, and ganache 2.0
        // NOTE this only works for a single failed transaction at a time.
```

File: packages/debugger/test/endstate.js (hunk 23 above)
Comment (gnidan) [reply packages/debugger/test/endstate.js:78]:
```suggestion
    assert.isDefined(txHash, "should have errored and set txHash");
```

File: packages/contract/lib/reason.js (hunk 12 above)
Comment (gnidan) [reply packages/contract/lib/reason.js]: Or maybe we don't need to borrow #4506 (assuming tests pass :)

File: packages/deployer/test/ens.js [hunk 29]
```diff
@@ -20,25 +20,26 @@ let ganacheOptions,
 describe("ENS class", () => {
   before(() => {
     ganacheOptions = {
+      // note that when vmErrorsOnRPCResponse is true, strict instamine must be enabled (default)
```
Comment (gnidan) [inline packages/deployer/test/ens.js:23]:
```suggestion
      // note that when vmErrorsOnRPCResponse is true, `"eager"` instamine must be enabled (default)
```

File: packages/decoder/test/current/test/downgrade-test.js [hunk 30]
```diff
@@ -613,12 +613,15 @@ async function runErrorTestBody(mangledCompilations) {
...
```
Comment (gnidan) [inline packages/decoder/test/current/test/downgrade-test.js:617]:
```suggestion
  // NOTE we wrap this in `try`/`catch` to accommodate new eth_call behavior
  // see https://github.com/trufflesuite/ganache/issues/1496
  try {
```

File: packages/decoder/test/current/test/wire-test.js (hunk 6 above)
Comment (gnidan) [inline packages/decoder/test/current/test/wire-test.js:1086]:
```suggestion
      // NOTE we wrap this in `try`/`catch` to accommodate new eth_call behavior
      // see https://github.com/trufflesuite/ganache/issues/1496
      try {
```

**Review by gnidan — APPROVED** [review]

File: packages/debugger/test/data/global.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (haltman-at) [reply packages/debugger/test/data/global.js]: OK I went and updated this particular one to London; the rest are still on Istanbul though.
